src/components/Models/LangModel/LangModel.tsx

```diff
@@ -0,0 +1,116 @@
+// THIS IS LANGUAGE MODEL POP UP
+'use client';
+
+import Image from 'next/image';
+
+import styles from './LangModel.module.css';
+
+// import { useParams } from 'next/navigation';
+// import {usePathname, useRouter} from '../../../navigation';
+import { usePathname, useRouter, useParams } from 'next/navigation';
+import { useTheme } from 'next-themes';
+
+export interface LangMenuItem {
+  code: string;
+  label: string;
+  name: string;
+  translation: string;
+  imageUrl: string;
+}
+
+const LangModel = (props: any) => {
+  const Languages: LangMenuItem[] = [
+    {
+      code: 'en',
+      label: 'EN',
+      name: 'English',
+      translation: 'English',
+      imageUrl: '/images/shared/language/en.svg',
+    },
+    {
+      code: 'es',
+      label: 'ES',
+      name: 'Spanish',
+      translation: 'Español',
+      imageUrl: '/images/shared/language/es.svg',
+    },
+    {
+      code: 'ko',
+      label: 'KO',
+      name: 'Korean',
+      translation: '한국어',
+      imageUrl: '/images/shared/language/ko.svg',
+    },
+    {
+      code: 'ng_hau',
+      label: 'NG/HAU',
+      name: 'Nigerian Hausa',
+      translation: 'Hausa',
+      imageUrl: '/images/shared/language/ng.svg',
+    },
+    {
+      code: 'ng_ibo',
+      label: 'NG/IBO',
+      name: 'Nigerian Igbo',
+      translation: 'Igbo',
+      imageUrl: '/images/shared/language/ng.svg',
+    },
+    {
+      code: 'ng_yor',
+      label: 'NG/YOR',
+      name: 'Nigerian Yoruba',
+      translation: 'Yorùbá',
+      imageUrl: '/images/shared/language/ng.svg',
+    },
+  ];
+
+  const pathname = usePathname();
+  const router = useRouter();
+
+  const { setTheme, theme } = useTheme();
+
+  const handleLanguageChange = (code: string) => {
+    const slipPathname = pathname.split('/').slice(2);
```

Comment:
Can `pathname` ever be null or undefined?  Would it be worth putting a null check here?

Reply:
It appears `npm run build` is failing possibly for this reason.